pytest.ini

```diff
@@ -0,0 +1,5 @@
+[pytest]
+markers =
+    serial: mark a test that requires more resources to run that are thus only suitable for serial run.
+    remote_required: mark a test that requires internet access.
+    gpu: mark a test that requires GPU.
```

Comment:
How about changing the affected tests (the one to be marked as require gpu) to take a `ctx` (or `gpu_ctx`) argument? Based on that, the markers could be inferred and there would be more control to parallelize GPU heavy tests across multiple GPUs

Reply:
(not implying this should be done in this PR, though of course it could)

Reply:
Sounds like a good idea. How would one skip GPU tests with that setup?

Reply:
Likely something like `https://docs.pytest.org/en/latest/example/markers.html#automatically-adding-markers-based-on-test-names` will work. In short, we need to add `def pytest_collection_modifyitems(items):` to `conftest.py`, which allows us to iterate over all tests (items) and modify them (ie. add a marker)

Reply:
I see. I’d like to give this more thought. Some immediate concerns are that this may seem somewhat magical and not as explicit, and that naming tests that require network access seem a bit awkward. May I follow this up later?

Reply:
Yes, I was not trying to imply this should be done in this PR